### PR TITLE
Homework-numbers

### PR DESCRIPTION
**Pull request by Owlullaby**
Trello card: [Homework-Section 2(Working with numbers)](https://trello.com/c/iNSp0BZQ/812-2-section-2-homework-working-with-numbers-subsection-21)

Description:
This is the homework that I have done for Section 2-subsection 21 (Homework: Working with numbers) where we make use of simple arithmetic on numbers to write a simple calculator script.

I make use of:
1. `gets.chomp` to take in user inputs for the first and the second number
2. `+` to add two the two numbers
3. `-` to subtract the first number with the second number
4. `*` to multiple the first number with the second number
5. `/` to divide the first number with the second number
6. `%` to get the first number mod the second number

### CODE REVIEW

File: section2-the-ruby-programming-language/numbers/calculator.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+num_1 = gets.chomp
+# asing for the second number input from the user
+puts "Enter the second number:"
+num_2 = gets.chomp
+
+# outputs the first number + - x / % by the second number
+puts "The first number added by the second number is #{num_1.to_f + num_2.to_f}"
+puts "The first number deducted by the second number is #{num_1.to_f - num_2.to_f}"
+puts "The first number multiplied by the second number is #{num_1.to_f * num_2.to_f}"
+puts "The first number divided by the second number is #{num_1.to_f / num_2.to_f}"
```
Comment (tlkiong) [inline section2-the-ruby-programming-language/numbers/calculator.rb:19]: Why do you need to do like this?
```
num_1.to_f / num_2.to_f
```

Comment (Owlullaby) [reply section2-the-ruby-programming-language/numbers/calculator.rb:19]: In order for arithmetic operations to work, we need it to be an integer or float. `get.chomp` gets input as string object so by using `.to_f`, we changes the string object to float when using division `/`.

Comment (tlkiong) [reply section2-the-ruby-programming-language/numbers/calculator.rb:19]: Why not `.to_i` ?

Comment (Owlullaby) [reply section2-the-ruby-programming-language/numbers/calculator.rb:19]: If using `.to_i` to convert to integer, when divided the output will be rounded up. Let's say `10.to_i / 8.to_i` will give us  `1`. Using the `.to_f` to convert to float, gives us an accurate answer compared to integer.

Comment (tlkiong) [reply section2-the-ruby-programming-language/numbers/calculator.rb:19]: what if 10/3?

Comment (Owlullaby) [reply section2-the-ruby-programming-language/numbers/calculator.rb:19]: `10.to_i / 3.to_i` gives us `3`
`10.to_f / 3.to_f` gives us `3.3333333333333335`
`(10.to_f / 3.to_f).round(2)` gives us `3.33` in two decimal places

Comment (tlkiong) [reply section2-the-ruby-programming-language/numbers/calculator.rb:19]:
```
10.to_i / 3.to_i gives us 3
```
why?

Comment (Owlullaby) [reply section2-the-ruby-programming-language/numbers/calculator.rb:19]: Because the output was rounded up to an integer?

Comment (tlkiong) [reply section2-the-ruby-programming-language/numbers/calculator.rb:19]: 10.to_i is the same as 10
3.to_i is the same as 3.

Logically (mathematically), 10/3 should be 3.333. Why is it resulting in 3 insteaD?

Comment (Owlullaby) [reply section2-the-ruby-programming-language/numbers/calculator.rb:19]: Because it is doing an integer division when using both integers.

Comment (tlkiong) [reply section2-the-ruby-programming-language/numbers/calculator.rb:19]: Why?
That don't make sense in maths no?

Comment (Owlullaby) [reply section2-the-ruby-programming-language/numbers/calculator.rb:19]: In integer division and modulus, the dividend is divided by the divisor into an integer quotient and a remainder. The integer quotient operation is referred to as integer division, and the integer remainder operation is the modulus. In the real world of data manipulation there are some things that are always handled in whole units or numbers (ex. integer data type). Thus when using two integers it results in an integer division.

**Review by tlkiong — APPROVED** [review]

**Review by hanyikoh — APPROVED** [review]

**Pull request merged**